### PR TITLE
add get_current_week helper

### LINKED ISSUE #104: add a get_current_week() function
perhaps @guga31bb's here? 

```r
get_current_week <- function() {

  # get season
  s <- dplyr::if_else(
    lubridate::month(lubridate::today("GMT")) >= 9,
    lubridate::year(lubridate::today("GMT")) ,
    lubridate::year(lubridate::today("GMT")) - 1
  )

  # Find the first Monday of September
  day1 <- lubridate::as_date(paste(s, "09-01", sep="-"))
  week1 <- as.POSIXlt(seq(day1, length.out=7, by="day"))
  monday1 <- week1[week1$wday == 1]

  # NFL season starts 4 days later
  first_game <- (monday1 + lubridate::days(3)) %>% lubridate::as_date()

  # current week number of NFL season is 1 + how many weeks have elapsed since first game
  # (ie first game is week 1)
  current_week <- 1 +
    lubridate::days(lubridate::today("America/New_York") - first_game) %>%
    .$day %>%
    magrittr::divide_by_int(7)

  if (current_week <= 1) current_week <- 1

  return(current_week)

}
```

### PR DESCRIPTION
**Pull request by tanho63**
Resolves #104 with two methods: one date-based and the other schedule-based. Defaulting to schedule-based. 

Also doing some file-renaming.

### CODE REVIEW

File: R/utils_date.R [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    current_week <- current_season[is.na(result),week]
+    return(current_week)
+  }
+
+  if(use_date){
+    # Find first Monday of September in current season
+    week1_sep <- as.POSIXlt(paste0(most_recent_season(),"-09-0",1:7), tz = "GMT")
+    monday1_sep <- week1_sep[week1_sep$wday == 1]
+
+    # NFL season starts 4 days later
```
Comment (guga31bb) [inline R/utils_date.R:72]: >     # NFL season starts 4 days later

This should say 3, this was a dumb typo in my comment

Comment (tanho63) [reply R/utils_date.R]: comment updated

**Comment by guga31bb** [comment]
One more (I think) comment:

During the day of first game of the season (prior to and during the game), this should return current_week = 1. But the non-date-based version will return NA because there's no games with not NA result, right? So we also need

`    if(all(is.na(current_season$result))) return(1)
`

**Comment by tanho63** [comment]
> During the day of first game of the season (prior to and during the game), this should return current_week = 1. But the non-date-based version will return NA because there's no games with not NA result, right? So we also need

the non-date-based version should return 1 up until the MNF game result is put in (the last game of that week) after which it should return 2, right?

**Comment by mrcaseb** [comment]
I would prefer the function without use_date argument. 

We could make it try to compute week from schedules and if that fails, e.g. because of internet connection, automatically compute date based.

**Comment by guga31bb** [comment]
> the schedule-based version should return 1 up until the MNF game result is put in (the last game of that week) after which it should return 2, right?

Yes, but my question is how it does this in the time period when all games are NA (because the first game of the season hasn't finished yet)

**Comment by tanho63** [comment]
OK, most issues should be resolved now. Uses min(week) rather than week[1] for extra robustness.

**Review by mrcaseb — APPROVED** [review]
Lgtm

**Pull request merged**